block/block.go

```diff
@@ -157,6 +159,9 @@ func (m *Manager) applyBlock(block *types.Block, commit *types.Commit, blockMeta
 
 	m.blockCache.Delete(block.Header.Height)
 
+	// check if the proposer needs to be changed and change it if that's the case
+	switchRole := m.Executor.UpdateProposerFromBlock(m.State, m.Sequencers, block)
```

Comment:
this state change is not committed, which can lead to DoS in case the node will have a reboot after sequencer change.
The state update needs to happen before `batch.Commit()`